src/Microsoft.DotNet.XHarness.WebAssembly/xunit-runner.cs

```diff
@@ -0,0 +1,349 @@
+	}
+}
+
+class WasmRunner : IMessageSink
+{
+	ITestFrameworkDiscoveryOptions DiscoveryOptions;
+	Discoverer discoverer;
+	ITestFrameworkExecutor executor;
+	ITestFrameworkExecutionOptions executionOptions;
+	List<ITestCase> testCases;
+	XunitProject project;
+
+	public WasmRunner (XunitProject project) {
+		this.project = project;
+
+		var assemblyFileName = "/" + project.Assemblies.First ().AssemblyFilename;
+		testCases = new List<ITestCase> ();
+
+		var assembly = Assembly.LoadFrom (assemblyFileName);
+		var assemblyInfo = new Xunit.Sdk.ReflectionAssemblyInfo (assembly);
+
+		var collectionBehaviorAttribute = assemblyInfo.GetCustomAttributes(typeof(CollectionBehaviorAttribute)).SingleOrDefault();
+		var testAssembly = new TestAssembly(assemblyInfo, null);
+
+		var collectionFactory = ExtensibilityPointFactory.GetXunitTestCollectionFactory(this, collectionBehaviorAttribute, testAssembly);
+
+		/*
+		object res = null;
+		res = Activator.CreateInstance (typeof (Xunit.Sdk.MemberDataDiscoverer), true, true);
+		Console.WriteLine ("DISC2: " + res);
+		*/
+
+		DiscoveryOptions = TestFrameworkOptions.ForDiscovery(null);
+		executionOptions = TestFrameworkOptions.ForExecution(null);
+
+		discoverer = new Discoverer (assemblyInfo, new NullSourceInformationProvider(), this, DiscoveryOptions, collectionFactory);
+
+		executor = new XunitTestFrameworkExecutor (assembly.GetName (), new NullSourceInformationProvider (), this);
+	}
+
+	static object ConvertArg (object arg, Type argType) {
+		if (arg == null || arg.GetType () == argType)
+			return arg;
+		// Not clear what conversions should be done
+		// IEnumerable<T> -> T[]
+		if (argType.IsArray) {
+			Type t = typeof (IEnumerable<>).MakeGenericType (argType.GetElementType ());
+			if (t.IsAssignableFrom (arg.GetType ())) {
+				var m = typeof (Enumerable).GetMethod ("ToArray").MakeGenericMethod (new Type [] { argType.GetElementType () });
+				return m.Invoke (null, new object [] { arg });
+			}
+		}
+		try {
+			if (arg != null && arg is IConvertible)
+				arg = Convert.ChangeType (arg, argType);
+		} catch {
+		}
+		return arg;
+	}
+
+	public virtual bool OnMessage (IMessageSinkMessage msg) {
+		if (msg is Xunit.Sdk.TestCaseDiscoveryMessage disc_msg) {
+			testCases.Add (disc_msg.TestCase);
+			return true;
+		}
+		//Console.WriteLine ("MSG:" + msg);		
+		/*
+		if (msg is Xunit.Sdk.DiagnosticMessage dmsg)
+			Console.WriteLine ("MSG:" + dmsg.Message);
+		else if (msg is Xunit.Sdk.TestCaseDiscoveryMessage disc_msg)
+			Console.WriteLine ("TEST:" + disc_msg.TestCase.DisplayName);
+		else
+			Console.WriteLine ("MSG:" + msg);
+		*/
+		return true;
+	}
+
+	public int nrun, nfail, nskipped, nfiltered;
+
+	public int Run () {
+		int state;
+		int tc_index;
+
+		Console.WriteLine ("Discovering tests...");
+
+		int n = 0;
+		while (n < 20) {
+			bool res = discoverer.Step ();
+			if (!res) {
+				break;
+			}
+		}
+
+		Console.WriteLine ("Running " +  testCases.Count + " tests...");
+		state = 3;
```

Comment:
state is just updated to 3 and not use anywhere else, am I right?

Reply:
removed